store/router.go

```diff
@@ -117,6 +119,10 @@ func (r *Router) Get(ctx context.Context, key []byte, cm commitments.CommitmentM
 
 // Put ... inserts a value into a storage backend based on the commitment mode
 func (r *Router) Put(ctx context.Context, cm commitments.CommitmentMode, key, value []byte) ([]byte, error) {
+	if r.readonly {
+		return nil, errors.New("read-only router does not support Put")
+	}
```

Comment:
What will this return to the client? Will the Put handler return a 500 with this error?

One other idea is to just disable the put route/handler entirely when in read mode. Have you considered that?

Reply:
Yes, this will return 500 with this error.

I've considered this approach and tried implementing this approach prior to the current PR. There's restriction from IRouter interface requiring the `[put](https://github.com/Layr-Labs/eigenda-proxy/blob/d124ef8bd80eb97f4feadd6901cc3947d6350874/store/router.go#L18)` handler, but I didn't think it made sense to remove `put` from the interface? 

Also thought about making a separate router type that contains exactly this super simple `put` handler, and create Router type based on the readonly config, but there was a lot of code repetition for all the other functions, and I didn't find a way to easily share functions between two child struct types (the regular `Router` and a `ReadOnly` 🤔  And thought about a wrapper type, but didn't think it was worth it at the end and a bit counter-intuitive . 

Please suggest and I'm happy to learn and redo:)

Reply:
Gotcha! This works I think. We should be returning a 400 instead though. I noticed we return 500s in a bunch of places where we shouldn't (see https://github.com/Layr-Labs/eigenda-proxy/issues/127 for eg)